rcsbapi/data/schema.py

```diff
@@ -415,28 +417,65 @@ def __construct_query_networkx(self, input_ids, input_type, return_data_list):
         return_data_name = [name.split('.')[-1] for name in return_data_list]
         attr_list = self.root_dict[input_type]
         attr_name = [id["name"] for id in attr_list]
         attr_kind = {attr["name"]: attr["kind"] for attr in attr_list}
 
-        if not all(key in attr_name for key in input_ids.keys()):
-            raise ValueError(f"Input IDs keys do not match attribute names: {input_ids.keys()} vs {attr_name}")
-
         if not all(item in self.node_index_dict.keys() for item in return_data_list):
             raise ValueError(f"Unknown item in return_data_list: {', '.join([str(item) for item in return_data_list if item not in self.node_index_dict.keys()])}")
-
-        for key, value in input_ids.items():
-            if attr_kind[key] == "SCALAR":
-                if not isinstance(value, str):
-                    raise ValueError(f"Input ID for {key} should be a single string")
-            elif attr_kind[key] == "LIST":
-                if not isinstance(value, list):
-                    raise ValueError(f"Input ID for {key} should be a list of strings")
-                if not all(isinstance(item, str) for item in value):
-                    raise ValueError(f"Input ID for {key} should be a list of strings") 
-
+        
+        if isinstance(input_ids, Dict):
+            if not all(key in attr_name for key in input_ids.keys()):
+                raise ValueError(f"Input IDs keys do not match attribute names: {input_ids.keys()} vs {attr_name}")
+
+            for key, value in input_ids.items():
+                if attr_kind[key] == "SCALAR":
+                    if not isinstance(value, str):
+                        raise ValueError(f"Input ID for {key} should be a single string")
+                elif attr_kind[key] == "LIST":
+                    if not isinstance(value, list):
+                        raise ValueError(f"Input ID for {key} should be a list of strings")
+                    if not all(isinstance(item, str) for item in value):
+                        raise ValueError(f"Input ID for {key} should be a list of strings") 
+ 
+        if isinstance(input_ids, List):
+            plural_types = [key for key, value in self.root_dict.items() 
+                for item in value 
+                    if item['kind'] == 'LIST']
+            if input_type not in plural_types:
+                raise ValueError(f"The input type '{input_type}' is not a plural type. Please use one of the following types: {', '.join(plural_types)}. If you want to use the input type '{input_type}', please input id as a dictionary")
+
+            for id in input_ids:
+                if (re.match(r'^(MA|AF)_.*_[0-9]+$', id) and 
+                    input_type in ["polymer_entities", "branched_entities", "nonpolymer_entities"]):
+                    attr_name = [id["name"] for id in attr_list]
+                elif (re.match(r'^(MA|AF)_.*$', id) and input_type == "entries"):
```

Comment:
This should be moved to the end of the CSM regex checks because this check will catch all types of ids